src/netlink-conky.h

```diff
@@ -0,0 +1,77 @@
+#include "config.h"
+
+#ifndef _CONKY_NETLINK_H_
+#define _CONKY_NETLINK_H_
+
+extern "C" {
+#include <netlink/attr.h>
+#include <netlink/cache.h>
+#include <netlink/genl/ctrl.h>
+#include <netlink/genl/family.h>
+#include <netlink/genl/genl.h>
+#include <netlink/msg.h>
+#include <netlink/route/link.h>
+
+#include <linux/ieee80211.h>
+#include <uapi/linux/nl80211.h>
+}
+
+#include "net_stat.h"
+
+#include <atomic>
+#include <functional>
+#include <map>
+#include <string>
+#include <variant>
+
+using nl_link_id = std::variant<int, char *, std::string>;
+
+/// @brief State of the callback.
+///
+/// Values smaller than 0 indicate an error.
+/// 0 means the callback is finished.
+/// Values greater than 0 indicate the callback isn't finished yet.
+enum class callback_state : int {
+  INVALID = -1,
+  DONE = 0,
+  IN_FLIGHT,
+};
+
+template <typename Data>
+class nl_task {
+  struct nl_cb *cb;
+  std::atomic<callback_state> state;
+  Data data;
+
+ public:
+  nl_task(int family, uint8_t request,
+          std::function<int(struct nl_msg *, Data *)> processor);
+  ~nl_task();
+};
```

Comment:
@brndnmtthws netlink uses callbacks to return data. Should I block main loop and wait for response or handle responses in the next interation? It doesn't affect complexity much, but I'm not sure:
- whether to pick data that's little stale or busy wait
  - wait only if `update_interval == 0`?

Reply:
Going with the approach where callbacks update data directly. By the time we get to drawing they might be finished already as `time iw wlan0 info` seems to run almost instantly.

For the initial version I'll make the calls blocking and then add support for blocking only if `update_interval == 0`.